`feat: `Alignment grouping

LaraCalic opened the pull request:
PLEASE FOLLOW THE CHECKLIST BELOW WHEN CREATING A NEW PULL REQUEST. THE
CHECKLIST IS FOR YOUR INFORMATION AND MUST BE REMOVED BEFORE SUBMITTING THE PULL
REQUEST.

## Checklist

- [x] Does the PR title follow the `<prefix>: title` scheme?

    The prefix must be one of:

    - `fix`: for a bugfix
    - `feat`: for a new feature
    - `refactor`: for an improvement of an existing feature
    - `perf`, `test`: for performance- or test-related changes
    - `docs`: for documentation-related changes
    - `build`, `ci`, `chore`: as appropriated for infrastructure changes

- [ ] Does this modify the public API as defined in `docs/versioning.rst`?

    - [ ] Does the PR title contain a `!` to indicate a breaking change?
    - [ ] Is there section starting with `BREAKING CHANGE:` in the PR body
          that explains the breaking change?

- [ ] Is the PR ready to be merged?

    - [x] If not: is it marked as a draft PR?

- [ ] Does this PR close an existing issue?

    - [ ] Is the issue correctly linked so it will be automatically closed
        upon successful merge (See closing keywords link in the sidebar)?

- The CI will initially report a missing milestone. One of the maintainers will
  handle assigning a milestone for book-keeping.

- An automated workflow will assign labels based on changed files, and whether
  or not reference files were changed. These do not have to be set manually.

- If you push updates, and you know they will be superceded later on, consider adding
  `[skip ci]` in the commit message. This will instruct the CI system not to run any
  jobs on this commit.

gagnonlg commented on Examples/Algorithms/Alignment/include/ActsExamples/Alignment/AlignmentAlgorithm.hpp:
```diff
@@ -24,6 +24,54 @@
+    }
+};
+
+class AlignmentAlgorithm {
+public:
+    void makeAlignmentGroup(const std::vector<AlignmentParameters>& alignments);
+
+private:
+    std::map<GeoIds, AlignmentGroup> alignmentGroups;  //  storage of AlignmentGroups 
+};
+
+ // function that will group al_par based on geoIds 
+
+void AlignmentAlgorithm::makeAlignmentGroup(const std::vector<AlignmentParameters>& alignments) {
+    for (const auto& alignmentPar : alignments) {
+        GeoIds superstructure_id = alignmentPar.getGeoIds();
+        if (alignmentGroups.find(superstructure_id) == alignmentGroups.end()) {
+            alignmentGroups[superstructure_id] = AlignmentGroup(superstructure_id);
+        }
+        AlignmentGroup& alignmentGroup = alignmentGroups[superstructure_id];
+        // @todo: each superstructure_id corresponds to a specific group,
+        // and for each group add a set of associated alignment parameters
+    }
+    // later: Define a function that will add a set of al_par (for each group)
+}
+
```
I think you can remove this -- 1) the AlignmentAlgorithm is already defined below and 2) the AlignmentGroup constructor is actually our makeAlignmentFunction so we do not need a separate one after all

LaraCalic replied:
Implemented.

andiwand committed:
test: Use deterministic RNG for tests (#2694)

apparently otherwise the random engine can be different on different platforms which I observed in https://github.com/acts-project/acts/pull/2625 with macos

at the same time I hardcoded the type for the `std::uniform_int_distribution`. apparently this can also make a difference for the random value
